networkx/algorithms/approximation/tests/test_steinertree.py

```diff
@@ -200,6 +210,39 @@ def test_remove_nonterminal_leaves(self):
 
         assert list(G) == [4, 5, 6]  # only the terminal nodes are left
 
+    def test_weighted(self):
```

Comment:
Nit: You can factor these out into parametrized tests to avoid having two very similar loops. And then you can parametrize by method and get rid of the loop entirely.